src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Statements.cs

```diff
@@ -152,6 +152,28 @@ private void EmitLoopExpression(Expression expr)
         private void EmitSwitchExpression(Expression expr, CompilationFlags flags)
         {
             SwitchExpression node = (SwitchExpression)expr;
+            
+            if (node.Cases.Count == 0)
+            {
```

Comment:
It appears that there is implicit assumption that all core ExpressionTypes are not reducible except for special 18 ExpressionTypes like ExpressionType.PostIncrementAssign and the like.

The part that all the core nodes are treated either as reducible or not based on ExpressionType makes me think we should not make switch reducible.